beacon_node/lighthouse_network/src/rpc/methods.rs

```diff
@@ -323,11 +314,14 @@ pub struct BlobsByRangeRequest {
 
     /// The number of slots from the start slot.
     pub count: u64,
+
+    /// maximum number of blobs in a single block.
+    pub max_blobs_per_block: usize,
```

Comment:
This value is the same on all requests, does it need to be appended to every request? Or can it be part of the handler struct

Reply:
Hi Lion, good question, the thing is:
https://github.com/sigp/lighthouse/blob/48dd3f385cd3ecde1baeb9135815f376d79b048a/beacon_node/network/src/sync/network_context.rs#L406-L415
the request is created here for example, it's outside of the `Rpc` `NetworkBehaviour`.
Also `BlobsByRangeRequest::max_blobs_requested(&self)` requires it, how would you envision storing `max_blobs_per_block`? 
My objective is for `Rpc` to be completely unware of `E: EthSpec` so that we can uncouple it from the rest of lighthouse and all its config values are passed dynamically.

Reply:
Actually, I take that back, if we increase the blob count we will have different requests with different limits at runtime. So okay with me to keep it on the request

Reply:
I don't think we should have added this field to `BlobsByRangeRequest`. It breaks the SSZ representation.